bundles/org.openhab.binding.insteon/src/main/java/org/openhab/binding/insteon/internal/handler/InsteonNetworkHandler.java

```diff
@@ -0,0 +1,124 @@
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseBridgeHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.State;
+import org.openhab.binding.insteon.internal.InsteonBinding;
+import org.openhab.binding.insteon.internal.config.InsteonNetworkConfiguration;
+import org.openhab.binding.insteon.internal.discovery.InsteonDeviceDiscoveryService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link InsteonNetworkHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Rob Nielsen - Initial contribution
+ */
+@NonNullByDefault
+public class InsteonNetworkHandler extends BaseBridgeHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(InsteonNetworkHandler.class);
+
+    private @Nullable InsteonNetworkConfiguration config;
+    private @Nullable InsteonBinding insteonBinding;
+    private @Nullable InsteonDeviceDiscoveryService insteonDeviceDiscoveryService;
+    private @Nullable ScheduledFuture<?> pollingJob = null;
+
+    public static ClassLoader classLoader = Thread.currentThread().getContextClassLoader();
+
+    public InsteonNetworkHandler(Bridge bridge) {
+        super(bridge);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+    }
+
+    @Override
+    public void initialize() {
+        logger.info("Starting Insteon bridge");
+        config = getConfigAs(InsteonNetworkConfiguration.class);
+
+        scheduler.execute(() -> {
+            insteonBinding = new InsteonBinding(this, config.getPort(), config.getDeviceStatisticsInterval(),
+                    config.getDeviceDeadCount(), config.getDevicePollInterval(), config.getAdditionalDevices(),
+                    config.getAdditionalFeatures(), config.getModemDbRetryTimeout());
+
+            updateStatus(ThingStatus.ONLINE);
+
+            if (!insteonBinding.startPollingWhenReady()) {
+                String msg = "Initialization failed, unable to start the Insteon bridge with the port '"
+                        + config.getPort() + "'.";
+                logger.error(msg);
+
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, msg);
+                return;
+            }
+
+            pollingJob = scheduler.scheduleWithFixedDelay(new Runnable() {
+                @Override
+                public void run() {
+                    insteonBinding.logDeviceStatistics();
+                }
+            }, 0, insteonBinding.getDeviceStatisticsInterval(), TimeUnit.SECONDS);
```

Comment:
```suggestion
            pollingJob = scheduler.scheduleWithFixedDelay(insteonBinding::logDeviceStatistics,
                0, insteonBinding.getDeviceStatisticsInterval(), TimeUnit.SECONDS);
```

Reply:
I coded this a bit differently, hopefully it's OK.